keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -228,14 +220,23 @@ data class ResourceActuationPaused(
   @JsonIgnore
   override val ignoreRepeatedInHistory = true
 
-  constructor(resource: Resource<*>, reason: String?, clock: Clock = Companion.clock) : this(
+  constructor(resource: Resource<*>, reason: String? = null) : this(
     resource.apiVersion,
     resource.kind,
     resource.id,
     resource.application,
     reason,
     clock.instant()
   )
+
+  constructor(resource: Resource<*>, reason: String? = null, timestamp: Instant) : this(
```

Comment:
Added to allow the synthetic events of this type to be added with a specific timestamp in `EventController`.